bindings/nodejs/index.js

```diff
@@ -19,9 +19,89 @@
 
 /// <reference types="node" />
 
+const { Writable, Readable } = require('node:stream')
+
+class ReadStream extends Readable {
+  constructor(operator, path, options) {
```

Comment:
`ReadStream` should be created by `Reader` so that users can control the reader behavior like `op.reader(options_for_reader).createReadStream(options_bypass_to_readable)`.

Reply:
I think this will be an extra method of the `reader`.

Then we will have:
`op.reader(path).createReaderStream(options_bypass_to_readable)` for controlling reader's behavior
and
`op.createReaderStream(path, options_bypass_to_readable)` for normal using
since we could not create a reader by just passing options, we will use the `with_xxx` style API.

Reply:
> we will use the `with_xxx` style API.

Is `with_xxx` style API widely used in nodejs? Maybe we can following the same API pattern like in python: `op.writer(path, options)`?

Reply:
It's okay. I'm just giving an example here. `reader(path, options` style API is more usual.

Reply:
Let's try writing some js code:

Version A:

```js
let r = op.readerSync(path, { version: "abc" }).createReaderStream({ highWaterMark: 5 * 1024});
```

Version B:

```js
let r = op.createReaderStream(path, { highWaterMark: 5 * 1024}).with_version("abc")
```

Reply:
After discussion, we will keep only version A.